### PR TITLE
Issue #5711: JavadocTagContinuationIndentation: no violation with block tag starting the line

### LINKED ISSUE #5711: JavadocTagContinuationIndentation: no violation with block tag starting the line
Identified at https://github.com/checkstyle/checkstyle/pull/5710#discussion_r180848374

````
$ cat TestClass.java
public class TestClass {
    /**
     * Returns the value represented by the specified string of the specified
     * type. Returns 0 for types other than float, double, int, and long.
     * @param text the string to be parsed.
     * @param type the token type of the text. Should be a constant of
     * {@link TokenTypes}.
     * @return the double value represented by the string argument.
     */
    public static double parseDouble(String text, int type) {
        return 0;
    }
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
<module name="JavadocTagContinuationIndentation">
  <property name="offset" value="4"/>
</module>
    </module>
</module>

$ java -jar checkstyle-8.9-all.jar -c TestConfig.xml TestClass.java
Starting audit...
Audit done.
````

I am expecting a violation at line 7 since it is a continuation of the `@param` tag and thus should be +4 indented. If any text is placed before `{@link` then a violation appears.

it would be good to provide example of Check work in xdoc - http://checkstyle.sourceforge.net/config_javadoc.html#JavadocTagContinuationIndentation

### LINKED ISSUE #8183: JavadocTagContinuationIndentation: doesn't report violations for html elements
https://checkstyle.org/config_javadoc.html#JavadocTagContinuationIndentation
1. Test.java:
```
gauravpunjabi@Gauravs-MacBook-Pro:~/Coding/Open Source/checkstyle$ javac Test.java 
gauravpunjabi@Gauravs-MacBook-Pro:~/Coding/Open Source/checkstyle$ cat -n Test.java 
     1  public class Test {
     2          /**
     3           * Some description here.
     4           *
     5           * @param a
     6           * Text description here. // violation
     7           * @param b
     8           * <b>Some description in bold</b> // OK
     9           */
    10          public void test(int a, int b) {}
    11  }
```
2. Config file:
```
gauravpunjabi@Gauravs-MacBook-Pro:~/Coding/Open Source/checkstyle$ cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <property name="charset" value="ASCII" />
    <property name="severity" value="error" />
    <property name="fileExtensions" value="java" />

    <module name="TreeWalker">
        <module name="JavadocTagContinuationIndentation" />
    </module>
</module>
```

3. Output:
```
gauravpunjabi@Gauravs-MacBook-Pro:~/Coding/Open Source/checkstyle$ java -jar target/checkstyle-8.32-SNAPSHOT-all.jar -c config.xml Test.java 
Starting audit...
[ERROR] /Users/gauravpunjabi/Coding/Open Source/checkstyle/Test.java:6: Line continuation have incorrect indentation level, expected level should be 4. [JavadocTagContinuationIndentation]
Audit done.
Checkstyle ends with 1 errors.
```

#### Expected
Line 8 should be reported as a violation, it is not correctly indented.

#### Explaination
JavadocTagContinutation does not report indentation violations for `HTML_ELEMENT`. As discussed in here https://github.com/checkstyle/checkstyle/pull/8002#discussion_r414565394 .
`HTML_ELEMENT` must also be properly intended to make the docs look good and easily readable.

### PR DESCRIPTION
**Pull request by wltan**
Fixes #5711, resolves #8183

Similar to #8002, the problem is with the logic behind how a violation is determined. Previously we only checked the `TEXT` node itself. If the `TEXT` is only a short whitespace (not enough indentation), we ignored it when in fact we need to check if there is a non-newline token (such as a `JAVADOC_INLINE_TAG `) immediately after it.

~~Regression: https://wltan.github.io/checkstyle-reports/2020-05-10/javadoctagcont-blocktag/
There are violations within Checkstyle itself which were not previously detected due to this issue. They can be found in the regression report as well. I've fixed those violations to use the correct indentation, in a separate commit so it's easier to tell from the main changes, and I'll squash it back into one commit if requested after review.~~

Internal violations have been merged, and manual regression report is deprecated.

Diff Regression projects: https://raw.githubusercontent.com/wltan/checkstyle-reports/master/auto-generate/pr-8177/projects-to-test-on.properties
Diff Regression config: https://raw.githubusercontent.com/wltan/checkstyle-reports/master/auto-generate/pr-8177/my_check.xml

### CODE REVIEW

**Comment by pbludov** [comment]
@wltan do you need any assistance to finish this PR?

**Comment by wltan** [comment]
> @wltan do you need any assistance to finish this PR?

@pbludov Yes, that would be much appreciated. Wercker is failing, which from experience likely means that there are checkstyle violations in other related projects like sevntu. It would be good if you could check the logs to pinpoint exactly what went wrong.

**Comment by pbludov** [comment]
> @pbludov Yes, that would be much appreciated. Wercker is failing, which from experience likely means that there are checkstyle violations in other related projects like sevntu.

This may be unrelated to this issue.

**Comment by pbludov** [comment]
GitHub, rebase

**Comment by wltan** [comment]
> GitHub, rebase

I have already rebased yesterday. Did something go wrong?

image

**Comment by rnveach** [comment]
Wercker failure:
>  Task :postgresql:checkstyleMain
[ant:checkstyle] [ERROR] /pipeline/source/.ci-temp/pgjdbc/pgjdbc/src/main/java/org/postgresql/Driver.java:208: Line continuation have incorrect indentation level, expected level should be 4. [JavadocTagContinuationIndentation]
[ant:checkstyle] [ERROR] /pipeline/source/.ci-temp/pgjdbc/pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java:2164: Line continuation have incorrect indentation level, expected level should be 4. [JavadocTagContinuationIndentation]

**Comment by pbludov** [comment]
The errors are
```
[ant:checkstyle] [ERROR] /pipeline/source/.ci-temp/pgjdbc/pgjdbc/src/main/java/org/postgresql/Driver.java:208: Line continuation have incorrect indentation level, expected level should be 4. [JavadocTagContinuationIndentation]
[ant:checkstyle] [ERROR] /pipeline/source/.ci-temp/pgjdbc/pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java:2164: Line continuation have incorrect indentation level, expected level should be 4. [JavadocTagContinuationIndentation]
```
https://github.com/pgjdbc/pgjdbc/blob/master/pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java#L2164
https://github.com/pgjdbc/pgjdbc/blob/master/pgjdbc/src/main/java/org/postgresql/Driver.java#L208

@romani What is the next step? Shoul @wltan send a tiny PR to pgjdbc or we need to clone this project to our repo?
Or both? I mean clone to make CI happy right now and send PR to pgjdbc to fix it later?

**Comment by pbludov** [comment]
PRs for jdbc repo: https://github.com/pgjdbc/pgjdbc/pull/2017, https://github.com/pgjdbc/pgjdbc/pull/2018

**Comment by pbludov** [comment]
GitHub, rebase

**Comment by romani** [comment]
@pbludov , please review https://github.com/checkstyle/checkstyle/pull/9190 it should unblock this PR.
@wltan , please move commit `minor: fix internal Checkstyle violations` to separate PR lets merge it quickly.

**Comment by pbludov** [comment]
GitHub, rebase

**Comment by romani** [comment]
@pbludov, do you plan review this PR ?

**Comment by pbludov** [comment]
report is good

**Review by pbludov — APPROVED** [review]
@wltan please take a look to https://github.com/checkstyle/checkstyle/pull/8177#issuecomment-761921889

**Comment by wltan** [comment]
#9192 opened, I will rebase this PR after it is merged

**Comment by romani** [comment]
Please rebase and drop minor commit. Please prepare PR for final review

**Comment by romani** [comment]
@nmancus1 , please help to review this PR.

**Comment by nmancus1** [comment]
@wltan please generate a new regression report, since it appears that the report in the PR description is quite old. We have a github action to do this now, see https://github.com/checkstyle/contribution/tree/master/checkstyle-tester#report-generation.

**Comment by nmancus1** [comment]
@wltan ping

**Comment by wltan** [comment]
Sorry, I was busy the past few days.

**Comment by wltan** [comment]
GitHub, generate report

**Comment by romani** [comment]
> Should I use a smaller set of test projects? 

better to use CI for this to economize your time.
please use less projects for execution.
You can execute report generation few times sequentially : part1, part2

**Comment by nmancus1** [comment]
@romani look at the projects file, it has OpenJDK10, 11, and 14 uncommented; this is why diff report is failing.  Here is a good one: https://gist.githubusercontent.com/nmancus1/ca61df3ebd259f03b7a6d4fd866520e1/raw/fca8f93b99256f34d3b12a667b6a3c384e0087fc/projects-to-test-on.properties if you want to use it.

**Comment by romani** [comment]
>we should just remove these repos for now as to not confuse contributors

Yes

**Comment by wltan** [comment]
> @romani look at the projects file, it has OpenJDK10 and 11 uncommented; this is why diff report is failing. Here is a good one: https://gist.githubusercontent.com/nmancus1/ca61df3ebd259f03b7a6d4fd866520e1/raw/fca8f93b99256f34d3b12a667b6a3c384e0087fc/projects-to-test-on.properties if you want to use it.
> 
> Edit: we should just remove these repos for now as to not confuse contributors, and only leave OpenJDK 9.

Ok, I have updated my config to match this

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTagContinuationIndentationCheck.java
Comment (nmancus1) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTagContinuationIndentationCheck.java:186]: Maybe a bit more detail here.

**Review by nmancus1 — CHANGES_REQUESTED** [review]
@wltan thanks for generating a new report.  Items:

**Comment by romani** [comment]
@wltan , please find time to finish PR.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTagContinuationIndentationCheck.java
Comment (wltan) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTagContinuationIndentationCheck.java:186]: Ok, added some description on what exactly the algorithm is checking for in the text.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadoctagcontinuationindentation/InputJavadocTagContinuationIndentationBlockTag.java
Comment (wltan) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadoctagcontinuationindentation/InputJavadocTagContinuationIndentationBlockTag.java:29]: > Please add some more complicated test cases, such as:
> 
> ```java
>     /**
>      * Javadoc
>      * @param args  // ok
>      * {@code this} line is not correctly indented // violation
>      *    {@code this}  // ok
>      * <pre>this line is not correctly indented</pre> // violation
>      */
> ```
> 
> Also, it would be good to show violations for html tags and other problem areas for sequential lines, such as:
> 
> ```java
>     /**
>      * Javadoc
>      * @param args  // ok
>      * {@code this} line is not correctly indented // violation
>      * {@code this} line is not correctly indented // violation
>      * <pre>this line is not correctly indented</pre> // violation
>      */
> ```

Ok, added these two cases along with the regression case. Do I need to come up with more?

Comment (wltan) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadoctagcontinuationindentation/InputJavadocTagContinuationIndentationBlockTag.java:29]: > There is a good test case with html list in diff report, too: https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/javadoctagcont-blocktag_2021170726/reports/diff/apache-ant/xref/home/runner/work/checkstyle/checkstyle/contribution/checkstyle-tester/repositories/apache-ant/src/main/org/apache/tools/ant/taskdefs/AbstractCvsTask.java.html#L749

One thing I found that could be slightly unintuitive is that although a violation is printed for line 749, no violation is generated for lines 750-761. This is because our `getAllNewlineNodes` only searches the 'top-level' `DetailNode` objects, and does not look inside tokens such as `HTML_ELEMENT` which could themselves contain newlines.

Below is the Javadoc tree structure for the `InputJavadocTagContinuationIndentationBlockTag` class. A `HTML_ELEMENT` begins at [84:7] and contains all the subsequent newline tokens such as [84:11], [85:11], [86:18].
```
    |--METHOD_DEF -> METHOD_DEF [98:4]
    |   |--MODIFIERS -> MODIFIERS [98:4]
    |   |   |--BLOCK_COMMENT_BEGIN -> /* [78:4]
    |   |   |   |--COMMENT_CONTENT -> *\r\n     * Long javadoc from regression test case (apache-ant).\r\n     *\r\n     * @param c the command line which will be configured\r\n     * if the commandline is initially null, the function is a noop\r\n     * otherwise the function append to the commandline arguments concerning\r\n     * <ul>\r\n     * <li>\r\n     * cvs package\r\n     * </li>\r\n     * <li>\r\n     * compression\r\n     * </li>\r\n     * <li>\r\n     * quiet or reallyquiet\r\n     * </li>\r\n     * <li>cvsroot</li>\r\n     * <li>noexec</li>\r\n     * </ul>\r\n      [78:6]
    |   |   |   |   `--JAVADOC -> JAVADOC [78:7]
    |   |   |   |       |--NEWLINE -> \r\n [78:7]
    |   |   |   |       |--LEADING_ASTERISK ->      * [79:0]
    |   |   |   |       |--TEXT ->  Long javadoc from regression test case (apache-ant). [79:6]
    |   |   |   |       |--NEWLINE -> \r\n [79:59]
    |   |   |   |       |--LEADING_ASTERISK ->      * [80:0]
    |   |   |   |       |--NEWLINE -> \r\n [80:6]
    |   |   |   |       |--LEADING_ASTERISK ->      * [81:0]
    |   |   |   |       |--WS ->   [81:6]
    |   |   |   |       |--JAVADOC_TAG -> JAVADOC_TAG [81:7]
    |   |   |   |       |   |--PARAM_LITERAL -> @param [81:7]
    |   |   |   |       |   |--WS ->   [81:13]
    |   |   |   |       |   |--PARAMETER_NAME -> c [81:14]
    |   |   |   |       |   |--WS ->   [81:15]
    |   |   |   |       |   `--DESCRIPTION -> DESCRIPTION [81:16]
    |   |   |   |       |       |--TEXT -> the command line which will be configured [81:16]
    |   |   |   |       |       |--NEWLINE -> \r\n [81:57]
    |   |   |   |       |       |--LEADING_ASTERISK ->      * [82:0]
    |   |   |   |       |       |--TEXT ->  if the commandline is initially null, the function is a noop [82:6]
    |   |   |   |       |       |--NEWLINE -> \r\n [82:67]
    |   |   |   |       |       |--LEADING_ASTERISK ->      * [83:0]
    |   |   |   |       |       |--TEXT ->  otherwise the function append to the commandline arguments concerning [83:6]    |   |   |   |       |       |--NEWLINE -> \r\n [83:76]
    |   |   |   |       |       |--LEADING_ASTERISK ->      * [84:0]
    |   |   |   |       |       |--TEXT ->   [84:6]
    |   |   |   |       |       |--HTML_ELEMENT -> HTML_ELEMENT [84:7]
    |   |   |   |       |       |   `--HTML_TAG -> HTML_TAG [84:7]
    |   |   |   |       |       |       |--HTML_ELEMENT_START -> HTML_ELEMENT_START [84:7]
    |   |   |   |       |       |       |   |--START -> < [84:7]
    |   |   |   |       |       |       |   |--HTML_TAG_NAME -> ul [84:8]
    |   |   |   |       |       |       |   `--END -> > [84:10]
    |   |   |   |       |       |       |--NEWLINE -> \r\n [84:11]
    |   |   |   |       |       |       |--LEADING_ASTERISK ->      * [85:0]
    |   |   |   |       |       |       |--TEXT ->   [85:6]
    |   |   |   |       |       |       |--HTML_ELEMENT -> HTML_ELEMENT [85:7]
    |   |   |   |       |       |       |   `--LI -> LI [85:7]
    |   |   |   |       |       |       |       |--LI_TAG_START -> LI_TAG_START [85:7]
    |   |   |   |       |       |       |       |   |--START -> < [85:7]
    |   |   |   |       |       |       |       |   |--LI_HTML_TAG_NAME -> li [85:8]
    |   |   |   |       |       |       |       |   `--END -> > [85:10]
    |   |   |   |       |       |       |       |--NEWLINE -> \r\n [85:11]
    |   |   |   |       |       |       |       |--LEADING_ASTERISK ->      * [86:0]
    |   |   |   |       |       |       |       |--TEXT ->  cvs package [86:6]
    |   |   |   |       |       |       |       |--NEWLINE -> \r\n [86:18]
    |   |   |   |       |       |       |       |--LEADING_ASTERISK ->      * [87:0]
    |   |   |   |       |       |       |       |--TEXT ->   [87:6]
    |   |   |   |       |       |       |       `--LI_TAG_END -> LI_TAG_END [87:7]
    |   |   |   |       |       |       |           |--START -> < [87:7]
    |   |   |   |       |       |       |           |--SLASH -> / [87:8]
    |   |   |   |       |       |       |           |--LI_HTML_TAG_NAME -> li [87:9]
    |   |   |   |       |       |       |           `--END -> > [87:11]
    |   |   |   |       |       |       |--NEWLINE -> \r\n [87:12]
    |   |   |   |       |       |       |--LEADING_ASTERISK ->      * [88:0]
    |   |   |   |       |       |       |--TEXT ->   [88:6]
    |   |   |   |       |       |       |--HTML_ELEMENT -> HTML_ELEMENT [88:7]
    |   |   |   |       |       |       |   `--LI -> LI [88:7]
    |   |   |   |       |       |       |       |--LI_TAG_START -> LI_TAG_START [88:7]
    |   |   |   |       |       |       |       |   |--START -> < [88:7]
    |   |   |   |       |       |       |       |   |--LI_HTML_TAG_NAME -> li [88:8]
    |   |   |   |       |       |       |       |   `--END -> > [88:10]
    |   |   |   |       |       |       |       |--NEWLINE -> \r\n [88:11]
    |   |   |   |       |       |       |       |--LEADING_ASTERISK ->      * [89:0]
    |   |   |   |       |       |       |       |--TEXT ->  compression [89:6]
    |   |   |   |       |       |       |       |--NEWLINE -> \r\n [89:18]
    |   |   |   |       |       |       |       |--LEADING_ASTERISK ->      * [90:0]
    |   |   |   |       |       |       |       |--TEXT ->   [90:6]
    |   |   |   |       |       |       |       `--LI_TAG_END -> LI_TAG_END [90:7]
    |   |   |   |       |       |       |           |--START -> < [90:7]
    |   |   |   |       |       |       |           |--SLASH -> / [90:8]
    |   |   |   |       |       |       |           |--LI_HTML_TAG_NAME -> li [90:9]
    |   |   |   |       |       |       |           `--END -> > [90:11]
    |   |   |   |       |       |       |--NEWLINE -> \r\n [90:12]
    |   |   |   |       |       |       |--LEADING_ASTERISK ->      * [91:0]
    |   |   |   |       |       |       |--TEXT ->   [91:6]
    |   |   |   |       |       |       |--HTML_ELEMENT -> HTML_ELEMENT [91:7]
    |   |   |   |       |       |       |   `--LI -> LI [91:7]
    |   |   |   |       |       |       |       |--LI_TAG_START -> LI_TAG_START [91:7]
    |   |   |   |       |       |       |       |   |--START -> < [91:7]
    |   |   |   |       |       |       |       |   |--LI_HTML_TAG_NAME -> li [91:8]
    |   |   |   |       |       |       |       |   `--END -> > [91:10]
    |   |   |   |       |       |       |       |--NEWLINE -> \r\n [91:11]
    |   |   |   |       |       |       |       |--LEADING_ASTERISK ->      * [92:0]
    |   |   |   |       |       |       |       |--TEXT ->  quiet or reallyquiet [92:6]
    |   |   |   |       |       |       |       |--NEWLINE -> \r\n [92:27]
    |   |   |   |       |       |       |       |--LEADING_ASTERISK ->      * [93:0]
    |   |   |   |       |       |       |       |--TEXT ->   [93:6]
    |   |   |   |       |       |       |       `--LI_TAG_END -> LI_TAG_END [93:7]
    |   |   |   |       |       |       |           |--START -> < [93:7]
    |   |   |   |       |       |       |           |--SLASH -> / [93:8]
    |   |   |   |       |       |       |           |--LI_HTML_TAG_NAME -> li [93:9]
    |   |   |   |       |       |       |           `--END -> > [93:11]
    |   |   |   |       |       |       |--NEWLINE -> \r\n [93:12]
    |   |   |   |       |       |       |--LEADING_ASTERISK ->      * [94:0]
    |   |   |   |       |       |       |--TEXT ->   [94:6]
    |   |   |   |       |       |       |--HTML_ELEMENT -> HTML_ELEMENT [94:7]
    |   |   |   |       |       |       |   `--LI -> LI [94:7]
    |   |   |   |       |       |       |       |--LI_TAG_START -> LI_TAG_START [94:7]
    |   |   |   |       |       |       |       |   |--START -> < [94:7]
    |   |   |   |       |       |       |       |   |--LI_HTML_TAG_NAME -> li [94:8]
    |   |   |   |       |       |       |       |   `--END -> > [94:10]
    |   |   |   |       |       |       |       |--TEXT -> cvsroot [94:11]
    |   |   |   |       |       |       |       `--LI_TAG_END -> LI_TAG_END [94:18]
    |   |   |   |       |       |       |           |--START -> < [94:18]
    |   |   |   |       |       |       |           |--SLASH -> / [94:19]
    |   |   |   |       |       |       |           |--LI_HTML_TAG_NAME -> li [94:20]
    |   |   |   |       |       |       |           `--END -> > [94:22]
    |   |   |   |       |       |       |--NEWLINE -> \r\n [94:23]
    |   |   |   |       |       |       |--LEADING_ASTERISK ->      * [95:0]
    |   |   |   |       |       |       |--TEXT ->   [95:6]
    |   |   |   |       |       |       |--HTML_ELEMENT -> HTML_ELEMENT [95:7]
    |   |   |   |       |       |       |   `--LI -> LI [95:7]
    |   |   |   |       |       |       |       |--LI_TAG_START -> LI_TAG_START [95:7]
    |   |   |   |       |       |       |       |   |--START -> < [95:7]
    |   |   |   |       |       |       |       |   |--LI_HTML_TAG_NAME -> li [95:8]
    |   |   |   |       |       |       |       |   `--END -> > [95:10]
    |   |   |   |       |       |       |       |--TEXT -> noexec [95:11]
    |   |   |   |       |       |       |       `--LI_TAG_END -> LI_TAG_END [95:17]
    |   |   |   |       |       |       |           |--START -> < [95:17]
    |   |   |   |       |       |       |           |--SLASH -> / [95:18]
    |   |   |   |       |       |       |           |--LI_HTML_TAG_NAME -> li [95:19]
    |   |   |   |       |       |       |           `--END -> > [95:21]
    |   |   |   |       |       |       |--NEWLINE -> \r\n [95:22]
    |   |   |   |       |       |       |--LEADING_ASTERISK ->      * [96:0]
    |   |   |   |       |       |       |--TEXT ->   [96:6]
    |   |   |   |       |       |       `--HTML_ELEMENT_END -> HTML_ELEMENT_END [96:7]
    |   |   |   |       |       |           |--START -> < [96:7]
    |   |   |   |       |       |           |--SLASH -> / [96:8]
    |   |   |   |       |       |           |--HTML_TAG_NAME -> ul [96:9]
    |   |   |   |       |       |           `--END -> > [96:11]
    |   |   |   |       |       |--NEWLINE -> \r\n [96:12]
    |   |   |   |       |       `--TEXT ->       [97:0]
    |   |   |   |       `--EOF -> <EOF> [97:5]
    |   |   |   `--BLOCK_COMMENT_END -> */ [97:5]
```

If this behavior is intended, then it should probably be better documented in the usage examples so that it is not mistaken for a bug. Otherwise, if this behavior is unintended then the fix is to modify `getAllNewlineNodes` to check for 'deep' `NEWLINE` tokens. In either case, I think it can be moved to a new issue so that this PR can focus on the bug it was originally intended to fix.

Comment (nmancus1) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadoctagcontinuationindentation/InputJavadocTagContinuationIndentationBlockTag.java:29]: > If this behavior is intended, then it should probably be better documented in the usage examples so that it is not mistaken for a bug. Otherwise, if this behavior is unintended then the fix is to modify getAllNewlineNodes to check for 'deep' NEWLINE tokens. In either case, I think it can be moved to a new issue so that this PR can focus on the bug it was originally intended to fix.

I agree on all counts, but it is good that we discovered this.  @wltan can you please create an issue?

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTagContinuationIndentationCheck.java
Comment (nmancus1) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTagContinuationIndentationCheck.java:186]: Resolved.

**Review by nmancus1 — APPROVED** [review]

**Comment by nmancus1** [comment]
@romani you are last reviewer left. I approve once we open an issue for https://github.com/checkstyle/checkstyle/pull/8177#discussion_r579386002

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadoctagcontinuationindentation/InputJavadocTagContinuationIndentationBlockTag.java
Comment (wltan) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadoctagcontinuationindentation/InputJavadocTagContinuationIndentationBlockTag.java:29]: Opened #9315

**Comment by romani** [comment]
This is false positive to my mind
https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/javadoctagcont-blocktag_2021192314/reports/diff/sevntu-checkstyle/index.html#A1

All violations that we have in sevntu project we need to send as PR, as it will be failure in CI after release.

**Comment by rnveach** [comment]
> All violations that we have in sevntu project we need to send as PR, as it will be failure in CI after release.

Why didn't CI catch this?
https://github.com/checkstyle/checkstyle/blob/master/wercker.yml#L175

**Comment by romani** [comment]
>This is false positive to my mind

This javadoc is do weird and wrong. I think it is ok to just fix this javadoc to move paragraph end to be before blocktags

**Review by romani — APPROVED** [review]
Code is good, report is good.

**Comment by romani** [comment]
CI validation works fine.

```
$ git diff
diff --git a/sevntu-checks/src/test/java/com/github/sevntu/checkstyle/checks/coding/AvoidModifiersForTypesCheckTest.java b/sevntu-checks/src/test/java/com/github/sevntu/checkstyle/checks/coding/AvoidModifiersForTypesCheckTest.java
index d821395..5ec55ae 100644
--- a/sevntu-checks/src/test/java/com/github/sevntu/checkstyle/checks/coding/AvoidModifiersForTypesCheckTest.java
+++ b/sevntu-checks/src/test/java/com/github/sevntu/checkstyle/checks/coding/AvoidModifiersForTypesCheckTest.java
@@ -37,7 +37,7 @@ import com.puppycrawl.tools.checkstyle.api.TokenTypes;
  * @author <a href="mailto:yasser.aziza@gmail.com">Yasser Aziza</a>
  * </p>
  */
-public class AvoidModifiersForTypesCheckTest extends AbstractModuleTestSupport {
+public class AvoidModifiersForTypesCheckTest extends                              AbstractModuleTestSupport {
 
     @Override
     protected String getPackageLocation() {


$ mvn -e -Pno-validations verify  -Dcheckstyle.skip=false -Dcheckstyle.version=8.41-SNAPSHOT -Dcheckstyle.configLocation=../../../checkstyle/checkstyle/config/checkstyle_checks.xml
...
[INFO] --- maven-checkstyle-plugin:3.1.1:check (default) @ sevntu-checks ---
[INFO] There are 2 errors reported by Checkstyle 8.41-SNAPSHOT with ../../../checkstyle/checkstyle/config/checkstyle_checks.xml ruleset.
[ERROR] src/test/java/com/github/sevntu/checkstyle/checks/coding/AvoidModifiersForTypesCheckTest.java:[40] (sizes) LineLength: Line is longer than 100 characters (found 109).
[ERROR] src/test/java/com/github/sevntu/checkstyle/checks/coding/AvoidModifiersForTypesCheckTest.java:[40,83] (whitespace) SingleSpaceSeparator: Use a single space to separate non-whitespace characters.

```

Reason is 
https://github.com/sevntu-checkstyle/sevntu.checkstyle/blob/81502e835d3772acf3ad53b2f306bc24b9ab88fa/sevntu-checks/suppressions.xml#L16-L17

**Comment by romani** [comment]
@wltan , thanks a lot for finishing PR, it took a while ... sorry that we did not focus on it enough to make it quick.

**Pull request merged**
